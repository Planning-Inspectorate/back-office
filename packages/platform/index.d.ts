import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
