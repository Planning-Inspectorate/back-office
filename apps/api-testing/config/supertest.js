// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
