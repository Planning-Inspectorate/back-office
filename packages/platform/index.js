export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
