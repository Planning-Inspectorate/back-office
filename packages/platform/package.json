{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
