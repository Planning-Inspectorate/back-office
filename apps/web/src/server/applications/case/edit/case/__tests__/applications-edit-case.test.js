import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { fixtureCases } from '../../../../../../../testing/applications/fixtures/cases.js';
import {
	fixtureRegions,
	fixtureSectors,
	fixtureSubSectors,
	fixtureZoomLevels
} from '../../../../../../../testing/applications/fixtures/options-item.js';
import { createTestEnvironment } from '../../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const nocks = () => {
	nock('http://test/').get('/applications/case-team').reply(200, {});
	nock('http://test/').get('/applications/sector').reply(200, fixtureSectors);
	nock('http://test/')
		.get(/\/applications\/3(.*)/g)
		.reply(200, fixtureCases[3]);
	nock('http://test/')
		.get(/\/applications\/4(.*)/g)
		.reply(200, fixtureCases[4]);
	nock('http://test/')
		.get('/applications/sector?sectorName=transport')
		.reply(200, fixtureSubSectors);
};

describe('applications edit', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	describe('Name', () => {
		const baseUrl = '/applications-service/case/3/edit/name';

		describe('GET /edit/name', () => {
			beforeEach(async () => {
				await request.get('/applications-service/case-team');
				nocks();
			});

			it('should render form', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('Save changes');
			});

			describe('When status is', () => {
				beforeEach(async () => {
					await request.get('/applications-service/case-team');
					nocks();
				});

				describe('Non-draft:', () => {
					it('should render the page with resumed data', async () => {
						const response = await request.get(baseUrl);
						const element = parseHtml(response.text);

						expect(element.innerHTML).toMatchSnapshot();
						expect(element.innerHTML).toContain(fixtureCases[3].title.slice(0, 20));
					});
				});
				describe('Draft:', () => {
					it('should not render the page', async () => {
						const response = await request.get('/applications-service/case/1/edit/name');
						const element = parseHtml(response.text);

						expect(element.innerHTML).toMatchSnapshot();
						expect(element.innerHTML).not.toContain('Save changes');
					});
				});
			});
		});
	});

	describe('Description', () => {
		describe('GET edit/description', () => {
			const baseUrl = '/applications-service/case/3/edit/description';

			beforeEach(async () => {
				await request.get('/applications-service/case-team');
				nocks();
			});

			it('should render page with resumed data', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain(fixtureCases[3].description.slice(0, 20));
			});
		});
	});

	describe('Project location', () => {
		const baseUrl = `/applications-service/case/3/edit/project-location`;

		beforeEach(async () => {
			await request.get('/applications-service/case-team');
			nocks();
		});

		describe('GET /edit/:caseId/project-location', () => {
			it('should display resumed data if the API returns something', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('London');
			});
		});
	});

	describe('Grid references', () => {
		const baseUrl = `/applications-service/case/3/edit/grid-references`;

		beforeEach(async () => {
			await request.get('/applications-service/case-team');
			nocks();
		});

		describe('GET /edit/:caseId/project-location', () => {
			it('should display resumed data if the API returns something', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('123456');
			});
		});
	});

	describe('Regions', () => {
		// const baseUrl = ( /!** @type {string} *!/ id) => `/applications-service/create-new-case/${id}/regions`;
		const baseUrl = (/** @type {string} */ id) => `/applications-service/case/${id}/edit/regions`;

		beforeEach(async () => {
			await request.get('/applications-service/case-team');
			nocks();
			nock('http://test/').get('/applications/region').reply(200, fixtureRegions);
		});

		describe('GET /edit/regions', () => {
			it('should render the page with checked options if the api returns something', async () => {
				const response = await request.get(baseUrl('3'));
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('value="london"\n                    checked');
				expect(element.innerHTML).toContain('value="yorkshire"\n                    checked');
			});
		});
	});

	describe('Zoom Level', () => {
		const baseUrl = (/** @type {string} */ id) =>
			`/applications-service/case/${id}/edit/zoom-level`;

		beforeEach(async () => {
			await request.get('/applications-service/case-team');
			nocks();
			nock('http://test/').get('/applications/zoom-level').reply(200, fixtureZoomLevels);
		});

		describe('GET edit/zoom-level', () => {
			it('should render the page with the checked option from the resumed data', async () => {
				const response = await request.get(baseUrl('3'));
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('value="city"\n                    checked');
			});
		});
	});

	describe('Team email', () => {
		const baseUrl = (/** @type {string} */ id) =>
			`/applications-service/case/${id}/edit/team-email`;

		beforeEach(async () => {
			await request.get('/applications-service/case-team');
			nocks();
			nock('http://test/').get('/applications/zoom-level').reply(200, fixtureZoomLevels);
		});

		describe('GET /edit/team-email', () => {
			it('should render the page with the resumed data if the api returns something', async () => {
				const response = await request.get(baseUrl('3'));
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('some@ema.il');
			});

			it('should render the page with no value inside the text input if api does not return resumed data', async () => {
				const response = await request.get(baseUrl('4'));
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).not.toContain("value='");
			});
		});
	});
});
