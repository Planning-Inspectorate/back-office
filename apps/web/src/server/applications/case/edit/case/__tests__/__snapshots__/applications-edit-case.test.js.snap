// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications edit Description GET edit/description should render page with resumed data 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter project description</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"description\\">Description of the project</label>
            <div id=\\"description-hint\\" class=\\"govuk-hint\\">for example, An offshore wind generating station of capacity up to 285
                MW</div>
            <textarea class=\\"govuk-textarea govuk-!-width-three-quarters\\" id=\\"description\\"
            name=\\"description\\" rows=\\"5\\" aria-describedby=\\"description-hint\\">Amet consectetur adipiscing elit sed do eiusmod tempor incididunt ut labore
                et dolore magna aliqua Ut enim ad minim veniam quis nostrud exercitation
                ullamco laboris nisi ut aliquip ex ea commodo consequat</textarea>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Grid references GET /edit/:caseId/project-location should display resumed data if the API returns something 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter grid references</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"geographicalInformation.gridReference.easting\\">Grid reference Easting</label>
            <div id=\\"geographicalInformation.gridReference.easting-hint\\"
            class=\\"govuk-hint\\">for example, 345678</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"geographicalInformation.gridReference.easting\\" name=\\"geographicalInformation.gridReference.easting\\"
            type=\\"text\\" value=\\"123456\\" aria-describedby=\\"geographicalInformation.gridReference.easting-hint\\">
        </div>
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"geographicalInformation.gridReference.northing\\">Grid reference Northing</label>
            <div id=\\"geographicalInformation.gridReference.northing-hint\\"
            class=\\"govuk-hint\\">for example, 345678</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"geographicalInformation.gridReference.northing\\" name=\\"geographicalInformation.gridReference.northing\\"
            type=\\"text\\" value=\\"987654\\" aria-describedby=\\"geographicalInformation.gridReference.northing-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Name GET /edit/name When role is: Case team should render form 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter project name</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"title\\">Name of the project</label>
            <div id=\\"title-hint\\" class=\\"govuk-hint\\">for example, Bridge over a very long canyon</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"title\\" name=\\"title\\" type=\\"text\\" value=\\"Title CASE/04\\" aria-describedby=\\"title-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Name GET /edit/name When role is: Inspector should NOT render the form 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <h1 class=\\"govuk-heading-xl\\">Sorry, there is a problem with your login</h1>
            <p class=\\"govuk-body\\">You are not permitted to access this URL.</p>
            <p class=\\"govuk-body\\"><a href=\\"https://has-appeal.herokuapp.com/help/contact\\" class=\\"govuk-link\\">Contact the Planning Inspectorate Customer Support</a> if
                the problem persists.</p>
        </div>
    </div>
</main>"
`;

exports[`applications edit Name GET /edit/name When status is Draft: should not render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <h1 class=\\"govuk-heading-xl\\">Sorry, there is a problem with the service</h1>
            <p class=\\"govuk-body\\">Try again later.</p>
            <p class=\\"govuk-body\\"><a href=\\"https://has-appeal.herokuapp.com/help/contact\\" class=\\"govuk-link\\">Contact the Planning Inspectorate Customer Support</a> if
                the problem persists.</p>
        </div>
    </div>
</main>"
`;

exports[`applications edit Name GET /edit/name When status is Non-draft: should render the page with resumed data 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter project name</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"title\\">Name of the project</label>
            <div id=\\"title-hint\\" class=\\"govuk-hint\\">for example, Bridge over a very long canyon</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"title\\" name=\\"title\\" type=\\"text\\" value=\\"Title CASE/04\\" aria-describedby=\\"title-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Project location GET /edit/:caseId/project-location should display resumed data if the API returns something 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter project location</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div id=\\"geographicalInformation.locationDescription-hint\\" class=\\"govuk-hint govuk-!-width-three-quarters\\">for example, approximately 8km off the coast of Kent, in areas surrounding
                Thanet Offshore Wind Farm</div>
            <textarea class=\\"govuk-textarea govuk-!-width-three-quarters\\"
            id=\\"geographicalInformation.locationDescription\\" name=\\"geographicalInformation.locationDescription\\"
            rows=\\"5\\" aria-describedby=\\"geographicalInformation.locationDescription-hint\\">London</textarea>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Regions GET /edit/regions should render the page with checked options if the api returns something 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Choose one or multiple regions</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div class=\\"govuk-checkboxes\\" data-module=\\"govuk-checkboxes\\">
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"london\\"
                    checked>
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames\\">London EN</label>
                </div>
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames-2\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"yorkshire\\"
                    checked>
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames-2\\">Yorkshire EN</label>
                </div>
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames-3\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"east\\">
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames-3\\">East EN</label>
                </div>
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames-4\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"north\\">
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames-4\\">North EN</label>
                </div>
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames-5\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"south\\">
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames-5\\">South EN</label>
                </div>
                <div class=\\"govuk-checkboxes__item\\">
                    <input class=\\"govuk-checkboxes__input\\" id=\\"geographicalInformation.regionNames-6\\"
                    name=\\"geographicalInformation.regionNames[]\\" type=\\"checkbox\\" value=\\"west\\">
                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"geographicalInformation.regionNames-6\\">West EN</label>
                </div>
            </div>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Team email GET /edit/team-email should render the page with no value inside the text input if api does not return resumed data 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter the project email (optional)</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label govuk-visually-hidden\\" for=\\"caseEmail\\">Team mail address</label>
            <div id=\\"caseEmail-hint\\" class=\\"govuk-hint\\">for example, NIEnquiries@planninginspectorate.gsi.gov.uk</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"caseEmail\\" name=\\"caseEmail\\" type=\\"text\\" aria-describedby=\\"caseEmail-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Team email GET /edit/team-email should render the page with the resumed data if the api returns something 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Enter the project email (optional)</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label govuk-visually-hidden\\" for=\\"caseEmail\\">Team mail address</label>
            <div id=\\"caseEmail-hint\\" class=\\"govuk-hint\\">for example, NIEnquiries@planninginspectorate.gsi.gov.uk</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"caseEmail\\" name=\\"caseEmail\\" type=\\"text\\" value=\\"some@ema.il\\" aria-describedby=\\"caseEmail-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications edit Zoom Level GET edit/zoom-level should render the page with the checked option from the resumed data 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h1 class=\\"govuk-heading-l\\"> Choose map zoom level</h1>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div class=\\"govuk-radios\\" data-module=\\"govuk-radios\\">
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"none\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName\\">None EN</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName-2\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"region\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName-2\\">Region EN</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName-3\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"country\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName-3\\">Country EN</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName-4\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"junction\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName-4\\">Junction EN</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName-5\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"county\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName-5\\">County EN</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"geographicalInformation.mapZoomLevelName-6\\"
                    name=\\"geographicalInformation.mapZoomLevelName\\" type=\\"radio\\" value=\\"city\\"
                    checked>
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"geographicalInformation.mapZoomLevelName-6\\">City EN</label>
                </div>
            </div>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;
